builder/git/git.go

```diff
@@ -0,0 +1,224 @@
+package git
+
+// This file just contains the Git-specific portions of sshd.
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"text/template"
+
+	"github.com/Masterminds/cookoo"
+	"github.com/Masterminds/cookoo/log"
+	"golang.org/x/crypto/ssh"
+)
+
+// PrereceiveHookTmpl is a pre-recieve hook.
+//
+// This is overridable. The following template variables are passed into it:
+//
+// 	.GitHome: the path to Git's home directory.
+var PrereceiveHookTpl = `#!/bin/bash
```

Comment:
If there were some way to keep this bash script as a separate file, yet have it be compiled in, we could have `shellcheck` validate it...